content/programming/joblib_low_overhead_persistence.md

```diff
@@ -73,54 +132,78 @@ Examples of new features:
 ['/tmp/test.pkl.lzma']
 >>> joblib.dump(obj, '/tmp/test.pkl.xz')  # xz
 ['/tmp/test.pkl.xz']
+```
 
-# Of course one can play with the compression level, but then the compressor
-# has to be given explicitly:
+Of course, one can play with the compression level, but then the compressor has
+to be given explicitly:
+```python
 >>> joblib.dump(obj, '/tmp/test.pkl.compressed', compress=('zlib', 6))
 ['/tmp/test.pkl.compressed']
 >>> joblib.dump(obj, '/tmp/test.compressed', compress=('lzma', 6))
 ['/tmp/test.pkl.compressed']
+```
 
-# The compressed files have a valid magic number so joblib can select
-# automatically the matching decompressor to read their content:
+Compressed files are valid regarding the compression method used. Joblib uses
+the Magic number at the beginning of the file to choose the right decompressor,
+making compressed pickle load transparent:
+```python
 >>> joblib.load('/tmp/test.compressed')
-[2 numpy arrays]
+[array([     0,      1,      2, ..., 999997, 999998, 999999]),
+ array([[ 1.,  1.,  1., ...,  1.,  1.,  1.],
+        ..., 
+        [ 1.,  1.,  1., ...,  1.,  1.,  1.]])]
 ```
 
-Examples of usage with file handles:
-```python
->>> import joblib
->>> data = "some data to persist"
+To conclude on those new exciting features, we can say a few words on file
+handles with contextlib. Indeed, all numpy arrays are stored in a
+single file, itself using standard compression formats and, as a consequence,
+__joblib now takes advantage of python `with` statement__ with objects
+implementing the buffer interface:
```

Comment:
I think that you mean "file-like objects", rather than the buffer interface, which is something completely different.